### PR TITLE
Add Button Global Component

### PR DESCRIPTION
**Pull request by jrojas28**
Some extra additions were done in order to improve styling and funcionality of this component:
- **Overall:**
-- Added Button Component
-- Added Button Component Tests

- **CSS Changes:**
-- Added _buttons mixin (required by Bootstrap Buttons)
-- Added _box-shadow mixin (required by Bootstrap Buttons)
-- Added _gradients mixin (required by Bootstrap Buttons)
-- Added _transitions mixin (required by Bootstrap Buttons)
-- Added Bootstrap Buttons 

- **Package Changes:**
-- Added Enzyme
-- Removed react-test-renderer

### CODE REVIEW

File: src/front/app/__test__/Button.test.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { shallow, render, mount } from 'enzyme';
+import { MemoryRouter, Link } from 'react-router-dom';
+import Button from '../components/globals/Button';
+
+const onClick = jest.fn();
+
+it('Should render itself', () => {
```
Comment (lrojas94) [inline src/front/app/__test__/Button.test.js]: it `renders` is descriptive enough.

File: src/front/app/__test__/Button.test.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  const wrapper = render(
+    <Button className="random">
+      <span>Im a Button!</span>
+    </Button>
+  );
+
+  expect(wrapper.attr('class')).toBe('btn btn-primary random');
+});
+
+it('Should render a Button by default', () => {
```
Comment (lrojas94) [inline src/front/app/__test__/Button.test.js]: you mean `button` in lowercase right? Button is the component

File: src/front/app/__test__/Button.test.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+it('Should call onClick once when clicked if onClick is provided.', () => {
+  const wrapper = shallow(
+    <Button className="random" onClick={onClick}>
+      <span>Im a Button!</span>
+    </Button>
+  );
+  wrapper.find('.btn').simulate('click');
+  expect(onClick).toHaveBeenCalled();
+});
```
Comment (lrojas94) [inline src/front/app/__test__/Button.test.js]: Missing test:
it: Adds btnType class whenever provided.

File: src/front/app/components/globals/Button.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      PropTypes.node,
+      PropTypes.instanceOf(Component),
+    ]),
+    href: PropTypes.string,
+    to: PropTypes.string,
+  };
+
+  static defaultProps = {
+    buttonType: 'primary',
+    children: 'Click Me!',
```
Comment (lrojas94) [inline src/front/app/components/globals/Button.js]: No, children is hard required, we don't want default values.

Comment (jrojas28) [reply src/front/app/components/globals/Button.js]: But you said specifically to add default values..?

**Pull request merged**